src/v/config/types.h

```diff
@@ -51,4 +51,28 @@ inline std::ostream& operator<<(std::ostream& os, const s3_url_style& us) {
     }
 }
 
+enum class schema_id_validation_mode {
```

Comment:
the `config::` system now fully owns this as part of its "external" interface with the outside work. the mirrored type in schema registry land still exists, but is fully internal and can have its own representation if it wants.

Reply:
What are your thoughts on having
```
v_schema_registry_config
v_schema_registry
   DEPENDS v_schema_registry_config
v_config
   DEPENDS v_schema_registry_config
```
It's the [pattern used in v_security](https://github.com/redpanda-data/redpanda/blob/dev/src/v/security/CMakeLists.txt)

Reply:
i don't have any strong opinions about either direction. the only downside i see to the subsys_config verison is that one has to be extra disciplined in separating things within subsys to avoid accidentally rebuilding the entire tree since config:: will depend on all _config libraries.

i went the direction in this PR to make that separation very clear. but actually having the duplicated enum is only out of being practical / time constraint. i'm not sure why at this point the config option can't just be a string with a couple helpers to convert.